client/src/routes/login/+page.svelte

```diff
@@ -1,46 +1,48 @@
 <script lang="ts">
-  import { z } from 'zod';
-  import { superForm, setMessage } from "sveltekit-superforms/client";
-    import { goto } from '$app/navigation';
+  import { z } from "zod";
+  import {
+    superForm,
+    superValidateSync,
+    setMessage,
+  } from "sveltekit-superforms/client";
+  import { goto } from "$app/navigation";
+  import type { ErrorResponse } from "bindings/ErrorResponse";
 
-  let form_data;
-
-  const userSchema = z.object({
+  const loginSchema = z.object({
     email: z.string().email(),
     password: z.string().min(8),
   });
 
-  const { form, errors, message, constraints, enhance } = superForm(
-    form_data,
+  const { form, message, errors, constraints, enhance } = superForm(
+    superValidateSync(loginSchema),
     {
       SPA: true,
-      validators: userSchema,
+      validators: loginSchema,
       onUpdate: async ({ form }) => {
-        console.log(form.valid);
         if (form.valid) {
-          setMessage(form, "Valid data!");
           const res = await fetch("http://localhost:6060/api/v1/users/login", {
             credentials: "include",
             method: "POST",
             headers: {
               "Content-Type": "application/json",
             },
-            body: JSON.stringify(form.data)
+            body: JSON.stringify(form.data),
           });
-          await goto("/");
+          if (res.status >= 400) {
+            const err: ErrorResponse = await res.json();
+            setMessage(form, err.error);
+          } else {
+            await goto("/");
+          }
         }
       },
-      onError({ result, message }) {
-        message.set(result.error.message);
-      },
     }
   );
-
 </script>
 
 <div class="login-container">
   <p>تسجيل الدخول</p>
-  {#if $message}<h3>{$message}</h3>{/if}
+  {#if $message}<h3 class="invalid">{$message}</h3>{/if}
```

Comment:
`message` is not an error message, shouldn't use `invalid` class, maybe add new class with green color and use that instead

Reply:
my idea was to show the error messages as status messages since form-level errors are removed after typing anything, explained [here](https://superforms.rocks/concepts/error-handling#form-level-and-array-errors).

> The form-level and array errors will be added and removed during [client-side validation](https://superforms.rocks/concepts/client-validation). If you would like a message to persist until the next submission, use a [status message](https://superforms.rocks/concepts/messages) instead.

Reply:
Alright, that works then, since it redirects to home page on successful submission anyways